inst/recorder/app.R

```diff
@@ -113,15 +113,40 @@ inputProcessors <- list(
   }
 )
 
+# Add in input processors registered by other packages.
+inputProcessors <- c(inputProcessors, shinytest::getInputProcessors())
+
 # Given an input value taken from the client, return the value that would need
 # to be passed to app$set_input() to set the input to that value.
 processInputValue <- function(value, inputType) {
-  if (is.null(inputProcessors[[inputType]]))
+  if (is.null(inputProcessors[[inputType]])) {
+    # For input with type "mypkg.foo", get "mypkg", and then try to load it.
+    # This is helpful in cases where the R session running `recordTest()` has
+    # not loaded the package with the input type. (There's a separate R session
+    # running the Shiny app.) See https://github.com/rstudio/learnr/pull/407 for
+    # more info.
+    pkg <- strsplit(inputType, ".", fixed = TRUE)[[1]][1]
+    tryLoadPackage(pkg)
+  }
+
+  # Check again if the input type is now registered.
+  if (is.null(inputProcessors[[inputType]])) {
     inputType <- "default"
+  }
 
   inputProcessors[[inputType]](value)
```

Comment:
Can we add an additional check that `is.function(inputProcessors[[inputType]])` ?

Reply:
I'll add this check on the registration side of things.